Src/java/cql-to-elm/src/test/resources/org/cqframework/cql/cql2elm/fhir/r401/TestInclude.cql

```diff
@@ -0,0 +1,170 @@
+library TestInclude
+
+using FHIR version '4.0.1'
+
+include FHIRHelpers version '4.0.1'
+
+codesystem "SNOMED": 'http://snomed.info/sct'
+codesystem "RXNORM": 'http://www.nlm.nih.gov/research/umls/rxnorm'
+
+code "aspirin 325 MG / oxycodone hydrochloride 4.84 MG Oral Tablet": '848768' from "RXNORM" display 'aspirin 325 MG / oxycodone hydrochloride 4.84 MG Oral Tablet'
+code "Acute hepatic failure": '197270009' from "SNOMED" display 'Acute hepatic failure'
+code "HIV Negative": '165815009' from "SNOMED" display 'HIV Negative'
+
+context Patient
+
+// Search Parameters
+
+define TestPractitionerSearch1:
+  [Practitioner: "?name" = 'ABC']
```

Comment:
Yes, that's expected at this point, I struggled with this quite a bit because whether you want to use code or search depends a great deal on what your target environment is, so we opted to make sure that the ELM supported both, but that the translator itself would remain largely silent on those types of rewrites, with the expectation that there would be a "planning" step at some point in the process that could do rewrites based on the target environment. For file-based processing, for example, codeProperty would be the way to go, but for an API-backed environment, codeSearch would need to be used. The Java engine already does quite a bit of this as part of the data access layer, so we are trying to ensure that the ELM supports rewrite with that planning in mind, but that the translator would basically stay out of it (other than the conceptual optimizations like date search).
Note also that the use of the question mark is intentionally undocumented behavior of the translator, used as a syntactic optimization as a placeholder for the planning step.
All that is to say, yes, it's expected that it still comes through as a codeProperty but with the indication that the user "suggested" a search parameter be used in the planning stage. Does that help?

Reply:
The intent of defining the Search in the ELM is to allow the rewrite planning to state searches formally in both the retrieve (using the XXXSearch attributes) and in the logic using the Search property accessor. I was originally planning to be more proactive about that in the translator, but over the course of implementing this feature, backed off of that in favor of the planning/rewrite approach. The tests were originally going to cover that whole scope within the translator, but are now just focused on verifying that the capabilities are functional from the ELM perspective. I will update the tests and documentation to make that more clear.